Options/Applicative/Types.hs

```diff
@@ -63,6 +63,7 @@ data ParseError
   | ShowHelpText
   | UnknownError
   | MissingError IsCmdStart SomeParser
+  | ExpectsArgError (Maybe String)
```

Comment:
This works, but there may be a better way to get the invariants right.

I could imaging a `ExpectsArg (String -> ParseError)` for instance, which would give the user more flexibility, but push the error message to the builder instead of renderer. One might change the builder to `String -> ParseError`, but that would break backwards compatibility.